src/seml/commands/migration.py

```diff
@@ -0,0 +1,166 @@
+import logging
+from typing import TYPE_CHECKING, Protocol
+from seml.database import get_collection
+from seml.settings import SETTINGS
+from seml.utils import s_if, smaller_than_version_filter
+from seml.utils.slurm import get_cluster_name
+
+
+if TYPE_CHECKING:
+    from pymongo.collection import Collection
+
+States = SETTINGS.STATES
+
+
+def migrate_collection(db_collection_name: str, skip: bool, backup: bool):
+    """
+    Migrate the given collection to be compatible with the current SEML version.
+
+    Parameters
+    ----------
+    db_collection_name : str
+        The name of the collection to migrate.
+    skip : bool
+        Whether to skip the migration.
+    backup : bool
+        Whether to create a backup of the collection before migrating.
+    """
+    from seml.console import prompt
+
+    collection = get_collection(db_collection_name)
+
+    if skip:
+        return
+
+    migrations = [migration_cls(collection) for migration_cls in _MIGRATIONS]
+    backed_up = False
+    for migration in migrations:
+        if not migration.requires_migration():
+            continue
+
+        if not migration.is_silent():
+            logging.warning(
+                f"The collection '{db_collection_name}' needs a {migration.name()} migration to work with newer SEML versions.\n"
+                'If you wish to make backups, do not proceed and call `seml --migration-backup <collection> <command>`.\n'
+                'To skip migration, use `seml --migration-skip <collection> <command>`.'
+            )
+            if not prompt('Do you want to proceed? (y/n)', type=bool):
+                logging.error('Aborted migration.')
+                exit(1)
+
+        if not backed_up and backup and not migration.is_silent():
+            backup_name = f'{db_collection_name}_backup'
```

Comment:
I'd suggest to make the backup-name a configurable template in `SETTINGS`, best with something that allows to add the current time (at least up to day) as well...

Reply:
Potentially even a qualifier for whether it was created automatically.